execute/README.md

```diff
@@ -0,0 +1,456 @@
+# Flux Engine Developer Guide
+
+This guide covers the Flux Engine design and related concepts.
+The engine includes:
+
+* Execution engine
+* Builtin sources and transformations
+* Memory layout and resource management
+
+## Concepts
+
+The following concepts are important to understanding the Flux engine and the operations that can be executed within the engine.
+This guide will reference these concepts with the assumption that the reader knows what is contained in this section.
+
+### Tables and Table Chunks
+
+In Flux, a **Table** is a set of ordered rows grouped together by a common **group key**.
+The **group key** is a set of key/value pairs where the value in the column with the same key name matches the value.
+In practical terms, this means that for any columns that are in the group key, the values in that table will all be the same.
```

Comment:
Suggestion: "The group key is a set of key/value pairs. The elements of the group key appear as columns in the table, with the element key appearing as the column name and the element value appearing as the value, constant across all rows.